plugins/auth-api/src/index.ts

```diff
@@ -62,16 +146,38 @@ class AuthCorePlugin implements AmplicationPlugin {
       staticPath
     );
 
-    return staticsFiles;
+    // create grants here, because here the DSG format this code to json file.
+    const grants =
+      context.entities && context.roles
+        ? createGrantsModule(
+            context.serverDirectories.srcDirectory,
+            context.entities,
+            context.roles
+          )
+        : null;
+
+    return grants ? [...staticsFiles, grants] : staticsFiles;
   }
 
   async afterCreateServerAuth(context: DsgContext) {
```

Comment:
I suggest moving the functions to another file, so in the index.ts file, you will have only the imports that you need and the register() events